### PR TITLE
feat: added context for ai; agents to use; module doc as example

### PR DESCRIPTION
**Pull request by Luxlorys**
# PR: Better AI Integration

## Why This Matters

AI-powered development tools work best when they understand your project's architecture, patterns, and conventions. Without proper context, agents produce generic code that requires manual fixes, slowing down development instead of accelerating it.

This PR adds structured context files that make agentic-based development **faster, streamlined, and effective**.

### Project Context (`CLAUDE.md`)

Central knowledge base that any AI agent reads first:
- Architecture overview (Routes → Handlers → Services → Repositories)
- CLI commands for code generation (`npm run generate:module`, `npm run generate:repository`)
- DI patterns with Awilix factory functions
- Coding conventions and file organization

**Result**: AI generates code that follows project patterns without manual corrections.

### Specialized Agents (`.claude/agents/`)

| Agent | Purpose |
|-------|---------|
| `ai-engineer.md` | LLM integrations, RAG systems, prompt engineering |
| `code-reviewer.md` | Automated code review with priority-based feedback |
| `database-architect.md` | Schema design, migrations, scaling strategies |
| `security-auditor.md` | OWASP compliance, auth flows, vulnerability detection |

**Result**: Domain-specific expertise on demand. Ask for a security review and get OWASP-aligned analysis, not generic suggestions.

### Slash Commands (`.claude/commands/`)

| Command | Purpose |
|---------|---------|
| `/generate-tests` | Comprehensive test suite generation (unit, integration, edge cases) |

**Result**: Consistent, repeatable workflows that follow project testing patterns.

### Module Documentation Example (`src/modules/message/README.md`)

Reference implementation showing how to document modules:
- API endpoints with request/response schemas
- Error responses
- Architecture diagrams
- Usage examples

**Result**: Template for documenting new modules consistently.

## Benefits

1. **Reduced Context Switching** - AI understands project structure immediately
2. **Consistent Output** - Generated code matches existing patterns
3. **Faster Onboarding** - New developers can use AI to understand the codebase
4. **Specialized Expertise** - Security, database, and AI engineering knowledge built-in
5. **Reproducible Workflows** - Slash commands ensure consistent processes

## Files Changed

```
+1,200 lines across 7 files

.claude/
├── agents/
│   ├── ai-engineer.md
│   ├── code-reviewer.md
│   ├── database-architect.md
│   └── security-auditor.md
└── commands/
    └── generate-tests.md

CLAUDE.md
src/modules/message/README.md
```

### CODE REVIEW

File: .claude/commands/generate-tests.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+- Component interaction testing
+- API integration with mocked responses
+- Service layer integration
+- End-to-end user workflows
+
+### Framework-Specific Tests
+
+- **React**: Component testing with React Testing Library
+- **Vue**: Component testing with Vue Test Utils
+- **Angular**: Component and service testing with TestBed
```
Comment (Max-Hirning) [inline .claude/commands/generate-tests.md]: Since this is a BE template, I don’t think we should add React, Vue, or Angular as options, so Claude doesn’t consider them as choices.

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+- Unit: `npm run test:unit`
+- Integration: `docker compose -f docker-compose.test.yml up` then `npm run test:int`
+
+## Important Conventions
+- Use factory functions, not classes
+- Register all DI dependencies with `addDIResolverName()`
+- Keep handlers thin - delegate to services
+- Validate inputs with Zod schemas
+- Use `@/` path alias for imports from src
+- **No inline comments** - Do not add comments after lines of code. JSDoc comments for functions/methods are allowed when they add meaningful context (e.g., security considerations, non-obvious behavior)
```
Comment (Max-Hirning) [inline CLAUDE.md:172]: I would also like to emphasize that this template does not use barrel files.
